### PR TITLE
feature(google-tag-manager): option to respect DNT header

### PR DESCRIPTION
**Pull request by williamchong007**
Hello I tried to implement a option to disable GTM according to DNT
please feel free to comment

### CODE REVIEW

**Comment by pi0** [comment]
DNT header is deprecated and not honored in many areas. https://www.zdnet.com/article/why-do-not-track-is-worse-than-a-miserable-failure

**Comment by pi0** [comment]
@williamchong007 Do you have any particular use case?

File: packages/google-tag-manager/plugin.js [hunk 1]
```diff
@@ -1,3 +1,14 @@
+// Detect Do Not Track settings
+<% if (options.respectDoNotTrack) { %>
+if (
+  window.doNotTrack === '1'
+  || navigator.doNotTrack === 'yes'
+  || navigator.doNotTrack === '1'
+  || navigator.msDoNotTrack === '1'
+  || window.external.msTrackingProtectionEnabled()
```
Comment (pi0) [inline packages/google-tag-manager/plugin.js]: `window.external && window.external.msTrackingProtectionEnabled && window.external.msTrackingProtectionEnabled()`

**Comment by williamchong007** [comment]
I was particularly dealing with https://github.com/EFForg/privacybadger on my website, which sets DNT=1 and I wanted to play nice
Major browsers have an option to set the DNT, I don' think it is deprecated, just not many would honour it.
This PR provides an option for users to actually start honouring them

**Pull request merged**